core/libdivecomputer.c

```diff
@@ -1013,13 +967,19 @@ static void verify_fingerprint(dc_device_t *device, device_data_t *devdata, cons
  */
 static void lookup_fingerprint(dc_device_t *device, device_data_t *devdata)
 {
+	uint32_t model, serial;
 	char *cachename;
 	struct memblock mem;
 
 	if (devdata->force_download)
 		return;
-	cachename = format_string("%s/fingerprints/%04x",
-		system_default_directory(), devdata->deviceid);
+
+	// Model hash and libdivecomputer 32-bit 'serial number' for the file name
+	model = calculate_string_hash(devdata->model);
+	serial = devdata->devinfo.serial;
+
+	cachename = format_string("%s/fingerprints/%04x.%u",
```

Comment:
Maybe factor this out into its own function, so that the fingerprint filenames can never run out of sync?